docs/python/generators/doxygen.py

```
@@ -236,7 +301,34 @@ def _parse_page_metadata():
+                for ce in class_examples:
+                    if 'src' not in ce and ce.get('srcs'):
+                        ce['src'] = Path(ce['srcs'][0])  # fallback default
+                        for s in ce['srcs']:
+                            try:
+                                src_text = Path(s).read_text(encoding='utf-8', errors='ignore')
+                                for name in top_names:
+                                    if name and re.search(r'\b' + re.escape(name) + r'\b', src_text):
+                                        ce['src'] = Path(s)
+                                        break
+                            except Exception:
+                                continue
```

Ruff (0.15.21)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)